### PR TITLE
Turboprop unit test 

### PR DESCRIPTION
**Pull request by xjjiang**
### Summary

1. added back `check_partials` in `test_electroprop` of `test_turboprop_model.py`.
2. in `PreHamiltonStandard` class, check whether inputs are valid.
3. in `PostHamiltonStandard` class, set propeller efficiencies to `0.0` if `power_coefficient` (CP) are very close to `0.0`. 
4. in Aviary docs, add a note for the case shaft power is zero.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by johnjasa — APPROVED** [review]
Nice changes, thanks Xun!

**Comment by jkirk5** [comment]
This solution raises an error if mission parameters like speed of sound, tip speed, shaft horsepower, etc. go negative - it's pretty easy to imagine a scenario where the optimizer to test a point slightly out of bounds and it will be extremely difficult to debug that if that behavior kills a run.

What's the typical way this is handled in complex openmdao problems? Should we add code to set specific parameters to an arbitrary small positive number if <= 0, then have the propeller builder add a constraint so the optimizer knows it shouldn't be doing that?

File: aviary/subsystems/propulsion/propeller/hamilton_standard.py [hunk 1]
```diff
@@ -524,9 +524,20 @@ def compute(self, inputs, outputs):
 
         # arbitrarily small number to keep advance ratio nonzero, which allows for static thrust prediction
         # NOTE need for a separate static thrust calc method?
-        vktas[np.where(vktas == 0.0)] = 1e-6
+        vktas[np.where(vktas <= 1e-6)] = 1e-6
         density_ratio = inputs[Dynamic.Mission.DENSITY] / RHO_SEA_LEVEL_ENGLISH
 
+        if diam_prop <= 0.0:
+            raise ValueError("Aircraft.Engine.PROPELLER_DIAMETER must be positive.")
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/propeller/hamilton_standard.py:531]:
```suggestion
            raise om.AnalysisError("Aircraft.Engine.PROPELLER_DIAMETER must be positive.")
```

File: aviary/subsystems/propulsion/propeller/hamilton_standard.py [hunk 2]
```diff
@@ -524,9 +524,20 @@ def compute(self, inputs, outputs):
...
+        if any(tipspd) <= 0.0:
+            raise ValueError("Dynamic.Mission.PROPELLER_TIP_SPEED must be positive.")
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/propeller/hamilton_standard.py:533]:
```suggestion
            raise om.AnalysisError("Dynamic.Mission.PROPELLER_TIP_SPEED must be positive.")
```

File: aviary/subsystems/propulsion/propeller/hamilton_standard.py [hunk 3]
```diff
@@ -524,9 +524,20 @@ def compute(self, inputs, outputs):
...
+        if any(sos) <= 0.0:
+            raise ValueError("Dynamic.Mission.SPEED_OF_SOUND must be positive.")
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/propeller/hamilton_standard.py:535]:
```suggestion
            raise om.AnalysisError("Dynamic.Mission.SPEED_OF_SOUND must be positive.")
```

File: aviary/subsystems/propulsion/propeller/hamilton_standard.py [hunk 4]
```diff
@@ -524,9 +524,20 @@ def compute(self, inputs, outputs):
...
+        if any(density_ratio) <= 0.0:
+            raise ValueError("Dynamic.Mission.DENSITY must be positive.")
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/propeller/hamilton_standard.py:537]:
```suggestion
            raise om.AnalysisError("Dynamic.Mission.DENSITY must be positive.")
```

File: aviary/subsystems/propulsion/propeller/hamilton_standard.py [hunk 5]
```diff
@@ -524,9 +524,20 @@ def compute(self, inputs, outputs):
...
+        if any(shp) < 0.0:
+            raise ValueError("Dynamic.Mission.SHAFT_POWER must be non-negative.")
```
Comment (jkirk5) [inline aviary/subsystems/propulsion/propeller/hamilton_standard.py:539]:
```suggestion
            raise om.AnalysisError("Dynamic.Mission.SHAFT_POWER must be non-negative.")
```

**Review by jkirk5 — CHANGES_REQUESTED** [review]

**Review by jkirk5 — APPROVED** [review]

**Pull request merged**